Node/core/lib/dialogs/EntityRecognizer.js

```diff
@@ -176,13 +176,34 @@ var EntityRecognizer = (function () {
                 score = Math.min(0.5 + (value.length / utterance.length), 0.9);
             }
             else {
-                var matched = '';
+                var matched = {};
                 tokens.forEach(function (token) {
                     if (value.indexOf(token) >= 0) {
-                        matched += token;
+                        if (!matched[token]) {
+                            matched[token] = 1;
```

Comment:
On [line 179][L179] we change `matched ` to an object and on [line 183][L183] we add a key-value pair to `matched` if the token was not already found in `matched`.


  [L179]: https://github.com/Microsoft/BotBuilder/pull/3702/files#diff-a5cbf2306473b89ce8f6eb849e7f09c5R179
  [L183]: https://github.com/Microsoft/BotBuilder/pull/3702/files#diff-a5cbf2306473b89ce8f6eb849e7f09c5R183